### PR TITLE
WIP: JSON Schema

### PR DESCRIPTION
**Pull request by oetiker**

### CODE REVIEW

**Comment by odeimaiz** [comment]
I still believe that "settings" entry is needed

**Comment by odeimaiz** [comment]
For the "simcore node", name is key at the same time. If we want to keep it as "name" I would make it more clear in the "description"

**Comment by oetiker** [comment]
lets call it key

**Comment by sanderegg** [comment]
Shouldn't there be a version entry as well?
And the actual value of an input/output? is this implicitly defined?

**Comment by mguidon** [comment]
The version is encoded into the tag of the image. Together with the labeling scheme for the images, that should also give us a link the the corresponding git hash

**Comment by sanderegg** [comment]
makes sense for the values.. just for the version of the scheme used by each image I'm not so sure. but anyway this looks very good.

**Comment by oetiker** [comment]
@sanderegg yes you are right ... we need a version ... since json schemas are identified by their 'url' is seems to be the custom that the version is encoded in the name ... (we will have to find a proper location for this file) in the meantime I have added a version number

**Comment by pcrespov** [comment]
Yes, I agree with @sanderegg . The version number should be  more explicit.
@mguidon docker-image tagging does indeed include version BUT it has itself a specific structure as well ``$(registry_name)/$(image_name):$(version)`` . Here we are using a key ``tag`` **only** for the last part.

**Comment by oetiker** [comment]
@pcrespov there is an actual version property forced to contain a semantic version number ... you want it more explicit? let's discuss Friday

**Comment by pcrespov** [comment]
@oetiker I believe I missed that. I saw ``tag`` keys with version numbers. If scheme enforces [semantic versioning](https://docs.npmjs.com/getting-started/semantic-versioning) then I fully support that.

**Comment by oetiker** [comment]
@mguidon the `_` IS allowed ... the `-` is not ... (at the moment)

**Comment by mguidon** [comment]
I see. Not very fluent with regex. I thought _ means to be case sensitive. Clancy solvers create files like 'output_1Hz.txt'

File: services/web/client/source/resource/qxapp/node-meta-v0.0.1.json [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+                        "type": {
+                            "type": "string",
+                            "pattern": "^(number|integer|boolean|string|data:([^/\\s]+/[^/\\s]+)$",
+                            "description": "data type expected on this input glob matching for data type is allowed",
+                            "examples": [
+                                "number",
+                                "boolean",
+                                "data:*/*",
+                                "data:text/*",
+                                "data:appliaction/json",
```
Comment (odeimaiz) [inline services/web/client/source/resource/qxapp/node-meta-v0.0.1.json]: appliaction -> application

File: services/web/client/source/resource/qxapp/node-meta-v0.0.1.json [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    "additionalProperties": false,
+    "required": [
+        "key",
+        "name",
+        "tag",
+        "description",
+        "authors",
+        "contact",
+        "inputs",
+        "outputs"
```
Comment (odeimaiz) [inline services/web/client/source/resource/qxapp/node-meta-v0.0.1.json]: @mguidon Correct me if I'm wrong but I think that settings are also needed

**Comment by oetiker** [comment]
merged into settings update

**Pull request closed without merging**